### PR TITLE
AnimeSama: Add support for multiple scan versions (#4156)

### LINKED ISSUE #4156: AnimeSama default source -> colorless
### Source name

AnimeSama

### Source language

French

### Describe your suggested feature

It seems that by default AnimeSama selects the color version of the manga (I only check One Piece). Thus it is not necessarily the most up to date version of the manga.
Would it be possible to take the colorless version instead?

### Other details

No

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/mihonapp/mihon/issues/new/choose).
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by CriosChan**
Closes #4156

Checklist:

- [X] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [X] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt [hunk 1]
```diff
@@ -62,19 +63,52 @@ class AnimeSama : ParsedHttpSource() {
...
-    override fun searchMangaSelector() = "a[href*='catalogue']"
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val entries: MutableList<SManga> = mutableListOf()
+
+        val document = response.asJsoup()
+
+        document.select(searchMangaSelector()).forEach { element ->
+            val fetchExistentSubMangas = GET(element.select("a").attr("href"), headers)
+            val page = client.newCall(fetchExistentSubMangas).execute()
```
Comment (AwkwardPeak7) [inline src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt]: this will result in n calls for n manga, please avoid this.

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt (hunk 1 above)
Comment (CriosChan) [reply src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt]: Would it be ok to limit the number of manga using the page system? I was also thinking about the possibility of making scanlators?

Comment (AwkwardPeak7) [reply src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt]: Pagination is an option but please avoid making individual calls for each manga in browse. 
I don't follow what you mean by scanlators? You can add them to chapters of that what you are asking

File: src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt [hunk 2]
```diff
@@ -100,16 +102,41 @@ class AnimeSama : ParsedHttpSource() {
     }
 
     override fun chapterListRequest(manga: SManga): Request {
```
Comment (AwkwardPeak7) [inline src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt]: request methods should ideally only return a request, you can make more requests in the parse method

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**